File: writers/lemonbar.lua
#!/usr/bin/env lua
require("luarocks.loader")

local sym = require("frmad.lib.sym").ascii
local processmetrics = require("frmad.lib.process_metrics")

local function statusline()
	local metrics = processmetrics()
	return string.format(
		"%%{l} %s%s %%{c}...%%{r} %s%s %s%s | %s %s%s | %s%s | %s%s | %s%s | %s%s\n",
		sym["clock"],
		os.date("%a %b %d, %Y | %H:%M:%S"),
		metrics.cpu.sym,
		metrics.cpu.val,
		metrics.cpu_temp.sym,
		metrics.cpu_temp.val,
		metrics.gpu.sym,
		metrics.cpu_temp.sym,
		metrics.gpu.val,
		metrics.mem.sym,
		metrics.mem.val,
		metrics.audio.sym,
		metrics.audio.val,
		metrics.net.sym,
		metrics.net.val,
		metrics.bat.sym,
		metrics.bat.val
	)
end

return {
	co = function()
		local user = os.getenv("USER")
		while true do
			local hout = assert(io.open("/tmp/frmad.lemonbar.out." .. user, "w"))
			hout:write(statusline())
			hout:close()
			coroutine.yield()
		end
	end,
	formatter = statusline,
	ri = 2,
}

File: m360/writers/lemonbar.py
import os
import io
from time import strftime
from m360.config.sym import ascii as sym
from m360.config.formats import formatvalue 
from asyncio import sleep as asleep

mget = lambda m, k: m[k] if k in m else k
fget = lambda m, k: formatvalue(k, m[k]) if k in m else "?"

def statusline(mtab: {}) -> str:
    return "%%{l} %s%s %%{c}...%%{r} %s%s %s%s | %s%s | %s%s | %s%s | %s%s | %s%s\n" % (
        sym["clock"],
        strftime("%a %b %d, %Y | %H:%M:%S"),
        sym["cpu"],
        fget(mtab, "cpu"),
        sym["cpu_temp"],
        fget(mtab, "cpu_temp"),
        mget(sym,  mget(mtab, 'gpu_id')+":gpu"),
        fget(mtab, mget(mtab, 'gpu_id')+":gpu_temp"),
        sym["mem"],
        fget(mtab, "mem"),
        sym["audio"],
        fget(mtab, "vol"),
        sym["net"],
        fget(mtab, "net_gateway"),
        mget(sym, "bat_"+fget(mtab, "battery_status").lower()),
        fget(mtab, "battery"),
    )

async def co(mtab: {}):
    user = os.getenv("USER")
    with io.open("/tmp/m360.lemonbar.out." + user, "w") as hout:
        while True:
            hout.write(statusline(mtab))
            hout.flush()
            await asleep(1) 
